### ISSUE TITLE
Filter nested attributes

### ISSUE THREAD

**Issue by rcrowe** [neomerx/json-api#47]
Given the following attributes:

``` json
"attributes": {
    "username": "vivalacrowe",
    "private": {
        "email": "hello@vivalacrowe.com",
        "name": "Rob"
    }
}
```

How do you filter to only get the email address. I've tried:

``` php
$params = new \Neomerx\JsonApi\Parameters\EncodingParameters(
    null,
    [
        'user' => ['private.email']
    ]
);
```

But this returns:

``` json
{
    "data": {
        "type": "user",
        "id": "12287",
        "links": {
            "self": "https:\/\/api.crowdcube.com\/users\/12287\/followers"
        }
    }
}
```

Filtering seems to only work on top level keys?

**Comment by neomerx** [comment]
@rcrowe 
Short answer: no it doesn't.
Longer answer: it's probably should not by the [spec](http://jsonapi.org/format) but we might to it.
JSON-API spec separates [filters](http://jsonapi.org/format/#fetching-filtering) and [field sets](http://jsonapi.org/format/#fetching-sparse-fieldsets). As the package supports both I'd prefer call the feature fieldsets.
As for fieldsets and [attributes](http://jsonapi.org/format/#document-resource-object-attributes) as a part of `fields` it looks that even for complex objects are attribute values but not attribute keys. However if it's not forbidden and not very slow we might try to add it.

**Comment by neomerx** [comment]
@rcrowe Do you want to start with this one? I can start with #48 as it might require deeper knowledge of this package.

**Comment by rcrowe** [comment]
Sure, as I'm new to the library want to point me in the right direction? :smile:

**Comment by neomerx** [comment]
Check out #48 first and this one :wink:

**Comment by neomerx** [comment]
@rcrowe I've added a [test](https://github.com/neomerx/json-api/tree/issue47/tests/Extensions/Issue47) for this issue. Can you please add filtering the way you like? You will need to change [this file](https://github.com/neomerx/json-api/blob/issue47/tests/Extensions/Issue47/CustomResourceObject.php).

**Comment by neomerx** [comment]
@rcrowe I haven't heard from you for some time, improved the code a bit and released it as [v0.5.6](https://github.com/neomerx/json-api/releases/tag/v0.5.6). If you have any further questions please don't hesitate to ask.
